lib/zendesk_apps_tools/deploy.rb

```diff
@@ -112,7 +118,18 @@ def check_job(job_id)
         sleep 3
       end
     rescue Faraday::Error::ClientError => e
-      say_error_and_exit e.message
+      delete_cache_exit_and_error(e.message)
+    end
+
+    private
+
+    def delete_cache_exit_and_error(error_message)
+      cache.delete(['app_id', 'subdomain', 'username'])
+      say_error_and_exit error_message
+    end
+
+    def connection(multipart = nil)
+      get_connection multipart
```

Comment:
Why not use `get_connection` directly everywhere?

Reply:
As discussed in real life. get_connection called with .post .get doesn't promote readability. 
(i.e. `get_connection.get('url')`)

The other alternative that we could use is ruby alias method name. We could call 
```
  alias_method 'name', 'full_name'
```
which is essentially doing what the method is doing.

Reply:
I like the `alias_method` suggestion

Reply:
I'll use alias_method :)